common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
+		this._setOptions(options);
+		if(this.client instanceof Realtime) {
+			this.client.connection.connectionManager.onAuthUpdated();
+		}
+	};
+
+	Auth.prototype.setClientId = function(clientId) {
+		if(clientId !== '*') {
+			if(this.clientId && this.clientId !== clientId) {
+				var msg = 'Unable to set incompatible clientId';
```

Comment:
> The library allows it, so you would get a failed connection instead of an exception which means the connection isn't touched.

But you said the library allows it.  So as far as I can tell, the following will happen.

1. User calls setOptions
2. New clientId is set
3. Transport is disconnected
4. Transport reconnects trying to resume and use new clientId, but Ably rejects it with an error code.
5. Client library moves to the `FAILED` state

So not sure why you say it's not touched.